RoddyCore/src/de/dkfz/roddy/core/Analysis.java

```diff
@@ -364,9 +364,11 @@ protected void executeRun(ExecutionContext context, boolean preventLoggingOnQuer
         String datasetID = context.getDataSet().getId();
         Exception eCopy = null;
         try {
+
+            context.analysis.getWorkflow().setupExecution(context);
             boolean contextRightsSettings = ExecutionService.getInstance().checkAccessRightsSettings(context);
             boolean contextPermissions = ExecutionService.getInstance().checkContextDirectoriesAndFiles(context);
-            boolean contextExecutability = context.checkExecutability();
+            boolean contextExecutability = context.analysis.getWorkflow().checkExecutability(context);
```

Comment:
Isn't this the same as before but just less readable and longer?

Reply:
Maybe, but I valued the readability of the client code lower than clarity of the semantics (single responsibility principle) of the ExecutionContext. The language is clearly to "execute the workflow in/with a specific context". Furthermore context is the much more abstract and volatile object. So I think the workflow is conceptually stronger than the context. As someone who is not deeply involved in the code I'd rather ask the workflow whether it is executible in a specific context (configs) than ask the context for some executibility (of what, anyway? it's not called checkWorkflowExecutibility()).

Does this make sense to you?

Reply:
Your explanation makes sense, but I whonder, if it makes sense to pull the workflow object from the analysis to the context.

Reply:
Yes, I see. In this particular case there was only this single usage. Generally using a facade is o.k., but ideally it would be used in a dedicated and appropriately named class, or not? I am also not a fan of call chains (isn't that something many OO programmers dislike?). But hiding call chains is not the optimal remedy, I guess. Rather the design should bu thus one always knows where to get the data without chains (I know that's easy to say and hard to achieve :-).

If you wish I change it back. My changes here are meant as suggestions and if we discuss they may have already fulfilled an important function :-)